library/src/main/java/com/novoda/downloadmanager/lib/NotificationDisplayer.java

```diff
@@ -114,17 +114,15 @@ private void buildActionIntents(String tag, int type, Collection<DownloadBatch>
             hideIntent.putExtra(DownloadReceiver.EXTRA_BATCH_ID, batchId);
             builder.setDeleteIntent(PendingIntent.getBroadcast(context, 0, hideIntent, 0));
 
-            if (type == DownloadNotifier.TYPE_SUCCESS) {
-                builder.setAutoCancel(true);
+            builder.setAutoCancel(true);
 
-                String action = DownloadStatus.isError(batch.getStatus()) ? Constants.ACTION_LIST : Constants.ACTION_OPEN;
+            String action = DownloadStatus.isError(batch.getStatus()) ? Constants.ACTION_LIST : Constants.ACTION_OPEN;
 
-                Intent clickIntent = new Intent(action, uri, context, DownloadReceiver.class);
-                clickIntent.putExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_IDS, new long[]{batchId});
-                clickIntent.putExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_STATUSES, new int[]{batchStatus});
-                clickIntent.putExtra(DownloadReceiver.EXTRA_BATCH_ID, batchId);
-                builder.setContentIntent(PendingIntent.getBroadcast(context, 0, clickIntent, PendingIntent.FLAG_UPDATE_CURRENT));
-            }
+            Intent clickIntent = new Intent(action, uri, context, DownloadReceiver.class);
+            clickIntent.putExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_IDS, new long[]{batchId});
+            clickIntent.putExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_STATUSES, new int[]{batchStatus});
+            clickIntent.putExtra(DownloadReceiver.EXTRA_BATCH_ID, batchId);
```

Comment:
out of curiosity why do we have `EXTRA_NOTIFICATION_CLICK_DOWNLOAD_IDS` and `EXTRA_BATCH_ID` ? they both seem to have only the `batchId` of the current download